go/vt/tabletmanager/agent_rpc_actions.go

```diff
@@ -404,6 +410,79 @@ func (agent *ActionAgent) RunBlpUntil(ctx context.Context, bpl *blproto.BlpPosit
 // Reparenting related functions
 //
 
+// InitMaster breaks slaves replication, get the current MySQL replication
+// position, insert a row in the reparent_journal table, and returns
+// the replication position
+func (agent *ActionAgent) InitMaster(ctx context.Context) (myproto.ReplicationPosition, error) {
+	// first break the slaves, so anyone who may have been replicating
+	// before will stop. This is meant to catch misconfigured hosts.
+	if err := agent.MysqlDaemon.BreakSlaves(); err != nil {
+		return myproto.ReplicationPosition{}, err
+	}
+
+	// get the current replication position
+	rp, err := agent.MysqlDaemon.MasterPosition()
+	if err != nil {
+		return myproto.ReplicationPosition{}, err
+	}
+
+	// Set the server read-write, from now on we can accept real
+	// client writes. Note that if semi-sync replication is enabled,
+	// we'll still need some slaves to be able to commit
+	// transactions.
+	if err := agent.MysqlDaemon.SetReadOnly(false); err != nil {
+		return myproto.ReplicationPosition{}, err
+	}
+
+	// Change our type to master if not already
+	tablet, err := agent.TopoServer.GetTablet(agent.TabletAlias)
```

Comment:
Since the intention here is to unconditionally set Tablet.Type = MASTER (regardless of the contents returned by GetTablet), this would be more resilient with TopoServer.UpdateTabletFields. It's also one less error return to check.